src/App.tsx

```diff
@@ -1,26 +1,205 @@
-/* eslint-disable max-len */
+/* eslint-disable react-hooks/rules-of-hooks */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import {
+  addTodo,
+  deleteTodo,
+  getTodos,
+  toggleTodo,
+  USER_ID,
+} from './api/todos';
+import { ErrorMessage, FilterStatus, Todo } from './types/Todo';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Error } from './components/Error';
 
 export const App: React.FC = () => {
+  const [selectedTodo, setSelectedTodo] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState(ErrorMessage.DEFAULT);
+  const [isErrorVisible, setIsErrorVisible] = useState(false);
+  const [todoStatus, setTodoStatus] = useState(FilterStatus.ALL);
+  const [isLoading, setIsLoading] = useState(true);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [deleteId, setDeleteId] = useState<number[]>([]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  useEffect(() => {
+    setIsLoading(true);
+    setErrorMessage(ErrorMessage.DEFAULT);
+
+    getTodos()
+      .then((todos: Todo[]) => {
+        setSelectedTodo(todos);
+      })
+      .catch(() => {
+        setErrorMessage(ErrorMessage.LOAD);
+        setTimeout(() => {
+          setErrorMessage(ErrorMessage.DEFAULT);
+        }, 3000);
+      })
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  const visibleTodos = selectedTodo.filter(todo => {
+    switch (todoStatus) {
+      case FilterStatus.ACTIVE:
+        return !todo.completed;
+      case FilterStatus.COMPLETED:
+        return todo.completed;
+      default:
+        return true;
+    }
+  });
+
+  const handleFilterChange = (newTodoStatus: FilterStatus) => {
+    setTodoStatus(newTodoStatus);
+  };
+
+  useEffect(() => {
+    if (errorMessage !== ErrorMessage.DEFAULT) {
+      setIsErrorVisible(true);
+
+      const timer = setTimeout(() => {
+        setErrorMessage(ErrorMessage.DEFAULT);
+        setIsErrorVisible(false);
+      }, 3000);
+
+      return () => clearTimeout(timer);
+    }
+  }, [errorMessage]);
```

Comment:
let's move this useEffect in Error component and handle error and timer from there